tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -459,3 +462,314 @@ def test_partial_mpu_after_server_pod_restart(self):
+        LOGGER.info("Step 3: Download the copied objects & verify etags in degraded mode")
+        for bkt, obj in bkt_obj_dict.items():
+            resp = s3_test_obj.get_object(bucket=bkt, key=obj)
+            LOGGER.info("Get object response: %s", resp)
+            get_etag = resp[1]["ETag"]
+            assert_utils.assert_equal(put_etag, get_etag, "Failed in verification of Put & Get "
+                                                          f"Etag for object {obj} of bucket "
+                                                          f"{bkt}.")
+        LOGGER.info("Step 3: Downloaded copied objects & verify etags in degraded mode")
+        bkt_obj_dict1 = bkt_obj_dict.copy()
+        t_t = int(perf_counter_ns())
+        bucket_name = self.bucket_name
+        bkt_op = False
+        LOGGER.info("Create and list buckets")
+        bkt_obj_dict.clear()
+        for idx, bkt in enumerate(bkt_obj_dict):
+            bkt_obj_dict[bkt] = f"ha-obj{idx}-{t_t}"
+        LOGGER.info("Step 4: Copy object from the %s bucket to other buckets and verify copy "
+                    "object etags", bucket_name)
+        resp = self.ha_obj.create_bucket_copy_obj(event, s3_test_obj=s3_test_obj,
+                                                  bucket_name=bucket_name,
+                                                  object_name=self.object_name,
+                                                  bkt_obj_dict=bkt_obj_dict,
+                                                  file_path=self.multipart_obj_path, bkt_op=bkt_op)
+        assert_utils.assert_true(resp[0], f"Failed buckets are: {resp[1]}")
+        put_etag = resp[1]
+        LOGGER.info("Step 4: Successfully created multiple buckets and copied object from the %s "
+                    "bucket to other buckets and verified copy object etags", self.bucket_name)
+        LOGGER.info("Step 5: Restart server pod with replica method")
+        resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                       restore_method=self.restore_method,
+                                       restore_params={"deployment_name": self.deployment_name,
+                                                       "deployment_backup":
+                                                           self.deployment_backup,
+                                                       "num_replica": self.num_replica,
+                                                       "set_name": self.set_name},
+                                       clstr_status=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+        self.restore_pod = False
+        LOGGER.info("Step 5: Successfully restart server pod with replica method")
+        LOGGER.info("Step 6: Download the copied objects & verify etags.")
+        for bkt, obj in bkt_obj_dict.items():
```

Comment:
We can merge both the dict and verify the etags in loop instead of writing same steps again.